app/controllers/resources_controller.rb

```diff
@@ -11,6 +11,18 @@ def index
     options = params.permit(*allowed_params)
       .slice(*allowed_params).to_h.symbolize_keys
 
+    # If a maximum limit is configured, we need to verify that the requested
+    # limit does not exceed the maximum.
+    if conjur_config.api_resource_list_limit_max.positive? && !params[:count]
```

Comment:
This comment should be amended to also explain the `!params[:count]` portion. To explain that we don't want to apply a default limit if this is a count request.